components/eamxx/src/physics/iop_forcing/eamxx_iop_forcing_process_interface.cpp

```diff
@@ -0,0 +1,546 @@
+#include "physics/iop_forcing/eamxx_iop_forcing_process_interface.hpp"
+
+#include "share/property_checks/field_within_interval_check.hpp"
+
+namespace scream
+{
+// =========================================================================================
+void IOPForcing::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ekat::units;
+
+  m_grid = grids_manager->get_grid("Physics");
+  const auto& grid_name = m_grid->name();
+
+  m_num_cols = m_grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = m_grid->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  FieldLayout scalar2d     = m_grid->get_2d_scalar_layout();
+  FieldLayout scalar3d_mid = m_grid->get_3d_scalar_layout(true);
+  FieldLayout vector3d_mid = m_grid->get_3d_vector_layout(true,2);
+
+  constexpr int pack_size = Pack::n;
+
+  add_field<Required>("ps", scalar2d, Pa, grid_name);
+
+  add_field<Updated>("horiz_winds", vector3d_mid, m/s, grid_name, pack_size);
+  add_field<Updated>("T_mid", scalar3d_mid, K, grid_name, pack_size);
+
+  add_tracer<Updated>("qv", m_grid, kg/kg, pack_size);
+  add_group<Updated>("tracers", grid_name, pack_size, Bundling::Required);
+
+  // Sanity check that iop data manager is setup by driver
+  EKAT_REQUIRE_MSG(m_iop_data_manager,
+    "Error! IOPDataManager not setup by driver, but IOPForcing"
+    "being used as an ATM process.\n");
+
+  // Compute number of buffer views
+  const auto iop_nudge_tq = m_iop_data_manager->get_params().get<bool>("iop_nudge_tq");
+  const auto iop_nudge_uv = m_iop_data_manager->get_params().get<bool>("iop_nudge_uv");
+  if (iop_nudge_tq) m_buffer.num_1d_scalar_nlev += 2;
+  if (iop_nudge_uv) m_buffer.num_1d_scalar_nlev += 2;
+}
+// =========================================================================================
+void IOPForcing::
+set_computed_group_impl (const FieldGroup& group)
```

Comment:
The fix for the memory fails in https://github.com/E3SM-Project/scream/pull/3089 was adding back the `IOPForcing::set_computed_group_impl()` function which sets `m_num_tracers` before the buffer attempts to compute the number of bytes needed.

Reply:
Then, we could have also just done
```c++
m_num_tracers = get_group_in("tracers").m_info->size();
```
at the top of `requested_buffer_size_in_bytes`. But doing it in here is arguably cleaner.

Reply:
I think the num_tracers is only used for allocating and initializing the buffer in this AP, so maybe it would make more sense there and reduce code. I chose to bring back the function because it _could_ be used elsewhere in the class and then would make more sense IMO to be initialized in a group specific virtual function and not the buffer since I can't be guaranteed the buffer init stuff is called in the right spot for some other usage in the class.